### PR TITLE
Archive context on search

### PR DESCRIPTION
**Pull request by jamieparkinson**
This makes archive search results look something like this:
image

When the direct parent of a result is the root of the archive in question, rather than displaying a duplicate title in "part of" and "found in", it displays the reference number in "part of" (@DominiqueMarshall's solution!) to make the duplication at least a little more useful.

Other changes:

- autoformatting / stuff I had to do to satisfy the precommit hook after syncing with master, not sure how that happened.
- I made the search result work type icons vertically centred along with their labels, as per the designs

### CODE REVIEW

**Comment by jamieparkinson** [comment]
Resolves https://github.com/wellcomecollection/wellcomecollection.org/issues/5752

File: catalogue/webapp/components/WorksSearchResult/WorksSearchResult.tsx [hunk 1]
```diff
@@ -73,6 +77,7 @@ function isPdfThumbnail(thumbnail): boolean {
 const WorkCard: FunctionComponent<Props> = ({ work }: Props) => {
   const productionDates = getProductionDates(work);
   const workTypeIcon = getWorkTypeIcon(work);
+  const archiveContext = getArchiveContext(work);
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorksSearchResult/WorksSearchResult.tsx:80]: can we not call it context so as not to conflict with react contexts?

**Comment by jamieparkinson** [comment]
I've put this behind a toggle so we can verify that the inclusion of `partOf` doesn't kick the API too hard

**Comment by jamieparkinson** [comment]
I accidentally merged https://github.com/wellcomecollection/wellcomecollection.org/pull/5914 into this rather than master 🙃

**Review by jamesgorrie — APPROVED** [review]

**Pull request merged**